src/cargo/core/resolver/dep_cache.rs

```diff
@@ -218,6 +222,7 @@ impl<'a> RegistryQueryer<'a> {
         parent: Option<PackageId>,
         candidate: &Summary,
         opts: &ResolveOpts,
+        first_minimal_version: bool,
     ) -> ActivateResult<Rc<(HashSet<InternedString>, Rc<Vec<DepInfo>>)>> {
         // if we have calculated a result before, then we can just return it,
         // as it is a "pure" query of its arguments.
```

Comment:
Trying to think whether we need to add `first_minimal_version` to the cache key. I think `first_minimal_version` is equivalent to `parent.is_none()` (` && -Zdirect-minimal-versions` but the `-Z` want change during resolution.) so redundant with information in the cache.

But either adding it to the cash key or adding a comment explaining why it's not needed would be appreciated.